### PR TITLE
Add required

### PR DESCRIPTION
**Pull request by nitro-neal**
Adding required

### CODE REVIEW

File: credential/exchange/builder_test.go [hunk 1]
```diff
@@ -144,7 +144,62 @@ func TestInputDescriptorBuilderProperties(t *testing.T) {
...
+				},
+			},
+		},
+	})
+
+	assert.NoError(t, err)
+	definition, err := builder.Build()
 	assert.NotEmpty(t, definition)
+	assert.NoError(t, err)
+	println(util.PrettyJSON(definition))
```
Comment (decentralgabe) [inline credential/exchange/builder_test.go:201]:
```suggestion
```

File: credential/exchange/model.go [hunk 2]
```diff
@@ -288,6 +288,7 @@ type Filter struct {
 	Type             string        `json:"type,omitempty"`
 	Format           string        `json:"format,omitempty"`
 	Properties       interface{}   `json:"properties,omitempty"`
+	Required         []string      `json:"required,omitempty"`
```
Comment (decentralgabe) [inline credential/exchange/model.go:291]: can we add `additionalProperties` which is a boolean value too?

Comment (nitro-neal) [reply credential/exchange/model.go:291]: ok yup I'll throw it in there

File: credential/exchange/builder_test.go [hunk 3]
```diff
@@ -144,7 +144,62 @@ func TestInputDescriptorBuilderProperties(t *testing.T) {
 	assert.NoError(t, err)
 	definition, err := builder.Build()
 	assert.NoError(t, err)
+	println(util.PrettyJSON(definition))
```
Comment (decentralgabe) [inline credential/exchange/builder_test.go:147]:
```suggestion
```

Comment (nitro-neal) [reply credential/exchange/builder_test.go]: lol sorry I got into merge hell

**Review by decentralgabe — APPROVED** [review]

**Pull request merged**